src/Input/Select/Select.noTypeChecking.test.tsx

```diff
@@ -46,17 +61,26 @@ describe('<Select /> can still work when receives non-string option children or
     const selectInput = queryByRole('combobox');
     userEvent.type(selectInput, '1');
     const filteredOptions = queryAllByTestId('option');
-    expect(filteredOptions.length).toEqual(1);
+    expect(filteredOptions.length).toEqual(3);
     expect(asFragment()).toMatchSnapshot();
   });
 
-  test('non-string defaultValue', async () => {
+  test('non-string and non-number defaultValue', async () => {
```

Comment:
it would be better if you can separate the numbers vs non-string and non-number tests to cover it's case since the test had already been there. we will be losing it's coverage if we replace it since they are 2 different cases